PR for M4 - *ONLY MERGE FOLLOWING EXPLICIT APPROVAL FROM RELEASE MANAGEMENT*

Kevsy opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:

Applies the Release Management instructions required for the M4 PR  (see #40 for details)

#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #40 

#### Special notes for reviewers:

Outstanding PRs: none at the time of this Pull Request. 

Note: there is one outstanding task from the API Readiness Checklist is the Test Result Statement - if possible I will try to create a lab test using the Test Definitions in this PR before the August 30th deadline. The Gherkin `.feature` has been checked as valid using a cucumber linter.

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.



```
docs

```

hdamker commented:
Thanks @Kevsy for the explicit warning in the title :-) ... have created the review issue for it.

AxelNennker commented:
Please review my ICM review https://github.com/camaraproject/IdentityAndConsentManagement/issues/189#issuecomment-2315026741

Kevsy commented:
> Please review my ICM review [camaraproject/IdentityAndConsentManagement#189 (comment)](https://github.com/camaraproject/IdentityAndConsentManagement/issues/189#issuecomment-2315026741)

Thanks @axel, responded on the ICM review thread.

rartych commented on CHANGELOG.md:
```diff
@@ -4,7 +4,8 @@ NOTE:
 ## Table of contents
 
-- **[r1.2 - rc](#r12---rc)**
+- [r1.3][#r13]
```
Parantheses to make mardown link working:

```suggestion
- [r1.3](#r13)
```

rartych commented on code/API_definitions/simple-edge-discovery.yaml:
```diff
@@ -2,7 +2,7 @@
 openapi: 3.0.3
 info:
  title: Simple Edge Discovery API
```
According to guidelines `# title without "API" in it, e.g. "Number Verification"`
```suggestion
 title: Simple Edge Discovery
```

Kevsy replied:
Thanks @rartych , committed

rartych commented:
In test definition the license text is not needed (or change DT with the right company)
  https://github.com/camaraproject/SimpleEdgeDiscovery/blob/93554d97fc83ac6f37438186c6b3496d413a01a4/code/Test_definitions/simple-edge-discovery.feature#L1-L20

Kevsy commented:
> In test definition the license text is not needed (or change DT with the right company)
> 
> https://github.com/camaraproject/SimpleEdgeDiscovery/blob/93554d97fc83ac6f37438186c6b3496d413a01a4/code/Test_definitions/simple-edge-discovery.feature#L1-L20

Thanks @rartych , now removed. Also corrected the number in the last test (copy-and-paste error)

tanjadegroot commented:
Please see https://github.com/camaraproject/ReleaseManagement/issues/89 on how to update the Test result statement in the API readiness checklist for this Fall24 meta-release.

rartych commented:
The text from https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#appendix-a-infodescription-template-for-device-identification-from-access-token is __recommended__ to be used for APIs  which use the `device` object.
Since `device` object is not used in this specification it can bring confusion.
I propose to modify the suggested text (lines 115-137) by replacing  _`device` object_ with something like _device identifying headers_ as defined from line 55 on
https://github.com/camaraproject/SimpleEdgeDiscovery/blob/d2e0953045776e7288d5925842761e10e7a3dd6c/code/API_definitions/simple-edge-discovery.yaml#L115-L137

The text was also modified in API specifications where only phoneNumber was used as the identifier.

Small typo in line 53: access token where **implemented** by the operator.

Kevsy commented:
@rartych thanks, changed in https://github.com/camaraproject/SimpleEdgeDiscovery/pull/41/commits/9fd8fa9a72320d389561383ac5f84d1dbb984b64

rartych commented:
@Kevsy please add again @camaraproject/release-management_maintainers as reviewer - it disappeared somehow.

rartych commented:
One more point - as it is mentioned in the text on 3-legged token the error response UNIDENTIFIABLE_DEVICE needs to be added to `Generic422`

 ```
           GENERIC_422_UNIDENTIFIABLE_DEVICE:
              description: The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token
              value:
                status: 422
                code: UNIDENTIFIABLE_DEVICE
                message: The device cannot be identified.
```

Kevsy commented:
> One more point - as it is mentioned in the text on 3-legged token the error response UNIDENTIFIABLE_DEVICE needs to be added to `Generic422`
> 
> ```
>           GENERIC_422_UNIDENTIFIABLE_DEVICE:
>              description: The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token
>              value:
>                status: 422
>                code: UNIDENTIFIABLE_DEVICE
>                message: The device cannot be identified.
> ```

ACK, will do this afternoon and let you know.

Kevsy commented:
> > One more point - as it is mentioned in the text on 3-legged token the error response UNIDENTIFIABLE_DEVICE needs to be added to `Generic422`
> > ```
> >           GENERIC_422_UNIDENTIFIABLE_DEVICE:
> >              description: The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token
> >              value:
> >                status: 422
> >                code: UNIDENTIFIABLE_DEVICE
> >                message: The device cannot be identified.
> > ```
> 
> ACK, will do this afternoon and let you know.

Hi @rartych , now fixed in https://github.com/camaraproject/SimpleEdgeDiscovery/commit/334d0d5987ddb7344a9274138f456f2a4cef8773

tanjadegroot reviewed (changes requested):
Looks great ! a few notes, please check and decide.

Once done, LGTM from Release Management

tanjadegroot commented on README.md:
line 14: suggest to change "customer" to "developer".

hdamker replied:
We using (API) customer in many places, I would consider it here as correct.

Kevsy commented on CHANGELOG.md:
```diff
@@ -22,6 +23,24 @@ The below sections record the changes for each API version in each (pre-)release
+  - version in servers.url to v1
+  - API Readiness checklist updated 
+
+Full Changelog: https://github.com/camaraproject/SimpleEdgeDiscovery/compare/r1.2...r1.3
```
HI @tanjadegroot  -  unusual situation here is that the last public release (`r0.9.3 -rc` ) was in another repository (EdgeCloud), so I can't make a GitHub URL that directly compares `r1.3` and `r0.9.3 - rc`.

 The majority of the curated changes from 0.9/3 -rc  are noted in the section for r1.1 - rc, with some small changes in r1.2 and r.13. Shall I just add a sentence to explain that?

hdamker replied:
@Kevsy it is not about the GitHub URL but about the list of changes in the release notes for r1.3. As this is the public release, the list of changes should contain all changes since the last public release, not only the ones since the pre-release. Put yourself in the shoes of readers who are only reading the release notes of r1.3 to understand what has changed since the last public release they are using. And yes, that means to repeat the entries from r1.1-rc if they were not reverted later.

hdamker commented:
@Kevsy @camaraproject/simple-edge-discovery_codeowners Could you please address the remaining comments from @tanjadegroot? We need to get this PR merged and the release created.

Kevsy commented on code/API_definitions/simple-edge-discovery.yaml:
```diff
@@ -114,27 +114,48 @@ info:
+  - If the `device`  identifying header is not included in the request and the
+    device information cannot be derived from the 3-legged access token, nor
+    inferred by the operator if the request is made directly from the client,
+    the server will return a 422 `UNIDENTIFIABLE_DEVICE` error.
```
Thanks for catch, fixed in 472d8bc35ab587e7ce5865b0fac0a4b52c76a278

Kevsy commented:
@hdamker commit 472d8bc35ab587e7ce5865b0fac0a4b52c76a278 just made, including my suggestion for the  CHANGELOG question from Tanja, and the modified `.feature`. 

Shall I make another commit to change `developer` back to `customer` in the README?

hdamker commented:
> Shall I make another commit to change `developer` back to `customer` in the README?

The [template](https://github.com/camaraproject/Template_API_Repository/blob/main/README.md) uses currently "The API provides the customer with the ability to:", but the [Glossary](https://github.com/camaraproject/Commonalities/blob/main/documentation/Glossary.md) and [Project Charter](https://github.com/camaraproject/Governance/blob/7a3081602aeb25c9d0697d53de8568d8ea2e7716/ProjectCharter.md?plain=1#L93) is using "API consumer" which might be more correct.

hdamker commented:
@Kevsy Regarding the CHANGELOG please see my comment https://github.com/camaraproject/SimpleEdgeDiscovery/pull/41#discussion_r1750159861

hdamker reviewed (approved):
LGTM

Kevsy commented:
Thanks @hdamker and @tanjadegroot 

So...all ok for me to merge?

tanjadegroot commented:
> > Shall I make another commit to change `developer` back to `customer` in the README?
> 
> The [template](https://github.com/camaraproject/Template_API_Repository/blob/main/README.md) uses currently "The API provides the customer with the ability to:", but the [Glossary](https://github.com/camaraproject/Commonalities/blob/main/documentation/Glossary.md) and [Project Charter](https://github.com/camaraproject/Governance/blob/7a3081602aeb25c9d0697d53de8568d8ea2e7716/ProjectCharter.md?plain=1#L93) is using "API consumer" which might be more correct. In short: there is no normative term.

For the next meta-release, we need to cross-check all template/default and API related files for the following:
- avoid using customer: use API consumer or (application) End-user or whatever else is applicable
- avoid using Telco/operator/CSP: use API provider / API platform
- avoid using subscriber: use Device or End-user (defined as the application user) as applicable

These guidelines and more are captured here: https://wiki.camaraproject.org/x/UwESAg

tanjadegroot commented:
> Thanks @hdamker and @tanjadegroot
> 
> So...all ok to merge?

LGTM from RM - go for it

Kevsy commented:
> LGTM from RM - go for it

Thanks @tanjadegroot  - 

@crissancas please can you Approve the PR?
